### PR TITLE
Fix minimum phpunit version and allow v5

### PR DESCRIPTION
**Pull request by andig**
Follow-up of #111 and https://github.com/reactphp/http/pull/114

phpunit ^5.0 is required since as of 4.8.20 php 7.0 is not supported (see https://github.com/sebastianbergmann/phpunit/blob/4.8.32/ChangeLog-4.8.md)

Welcome to version hell :/

### CODE REVIEW

**Comment by clue** [comment]
> phpunit ^5.0 is required since as of 4.8.20 php 7.0 is not supported (see https://github.com/sebastianbergmann/phpunit/blob/4.8.32/ChangeLog-4.8.md)

This change was actually reverted in 4.8.21. In other words: This actually works perfectly fine currently and installs 4.8.35 on PHP 7 (at the time of writing this).

That being said, I agree that it makes perfect sense to support PHPUnit 5 as well :+1:

File: composer.json [hunk 1]
```diff
@@ -16,6 +16,6 @@
         }
     },
     "require-dev": {
-        "phpunit/phpunit": "~4.8"
+        "phpunit/phpunit": "^4.8.10|^5.0"
```
Comment (clue) [inline composer.json]: Should probably be `^4.8.10 || ^5.0`?

Also, PHPUnit 5 emits several warnings on PHP 7, can you address these here? See also #93 and #107.

Comment (clue) [inline composer.json]: Arguably, fixing the lowest version and supporting PHPUnit 5 *could* be two separate PRs, but I'll leave this up to you :+1:

**Comment by andig** [comment]
@clue was a bit lazy and didn't split. Both phpunit 4 and 5 tests are passing now.

**Comment by andig** [comment]
Once this is merged, https://github.com/reactphp/http/pull/114 should pass, too.

**Pull request merged**